job-server/src/main/scala/spark/jobserver/io/FileCacher.scala

```diff
@@ -50,10 +50,10 @@ trait FileCacher {
       override def accept(dir: File, name: String): Boolean = {
         val prefix = appName + "-"
         if (name.startsWith(prefix)) {
-          val suffix = name.substring(prefix.length)
-          (Pattern findFirstIn suffix).isDefined
```

Comment:
This is extra code that isn't needed. The focus here should be on binary names prefix matches. If a user submits a DELETE for binary "example", then I would assume all binaries in the user specified location will be deleted and here the prefix is determined as "example-"